packages/babel-plugin-transform-regenerator/src/regenerator/emit.ts

```diff
@@ -56,1288 +19,1353 @@ const Ep = Emitter.prototype;
-      // may seem a bit redundant. In theory, we could check here to
-      // determine if stmt knows how to handle its own label; for example,
-      // stmt happens to be a WhileStatement and so we know it's going to
-      // establish its own LoopEntry when we explode it (below). Then this
-      // LabeledEntry would be unnecessary. Alternatively, we might be
-      // tempted not to pass stmt.label down into self.explodeStatement,
-      // because we've handled the label here, but that's a mistake because
-      // labeled loops may contain labeled continue statements, which is not
-      // something we can handle in this generic case. All in all, I think a
-      // little redundancy greatly simplifies the logic of this case, since
-      // it's clear that we handle all possible LabeledStatements correctly
-      // here, regardless of whether they interact with the leap manager
-      // themselves. Also remember that labels and break/continue-to-label
-      // statements are rare, and all of this logic happens at transform
-      // time, so it has no additional runtime cost.
-      self.leapManager.withEntry(
-        new (leap.LabeledEntry as any)(after, stmt.label),
-        function () {
-          self.explodeStatement(path.get("body"), stmt.label);
-        },
-      );
-
-      self.mark(after);
-
-      break;
-
-    case "WhileStatement":
-      before = this.loc();
-      after = this.loc();
-
-      self.mark(before);
-      self.jumpIfNot(self.explodeExpression(path.get("test")), after);
-      self.leapManager.withEntry(
-        new (leap.LoopEntry as any)(after, before, labelId),
-        function () {
-          self.explodeStatement(path.get("body"));
-        },
-      );
-      self.jump(before);
-      self.mark(after);
-
-      break;
-
-    case "DoWhileStatement":
-      const first = this.loc();
-      const test = this.loc();
-      after = this.loc();
-
-      self.mark(first);
-      self.leapManager.withEntry(
-        new (leap.LoopEntry as any)(after, test, labelId),
-        function () {
-          self.explode(path.get("body"));
-        },
-      );
-      self.mark(test);
-      self.jumpIf(self.explodeExpression(path.get("test")), first);
-      self.mark(after);
-
-      break;
-
-    case "ForStatement":
-      head = this.loc();
-      const update = this.loc();
-      after = this.loc();
-
-      if (stmt.init) {
-        // We pass true here to indicate that if stmt.init is an expression
-        // then we do not care about its result.
-        self.explode(path.get("init"), true);
-      }
+    if (path.isExpression()) return self.explodeExpression(path, ignoreResult);
 
-      self.mark(head);
+    switch (node.type) {
+      case "Program":
+        return (path.get("body") as NodePath[]).map(
+          // @ts-expect-error FIXME: The second parameter of `explodeStatement` is `t.Identifier`
+          self.explodeStatement,
+          self,
+        );
```

Comment:
![image](https://github.com/user-attachments/assets/657cfda2-048a-4de1-9330-086377a3defa)
I really don't understand why no error is thrown.

Reply:
Probably because Babel 8 has refined the arguments typings?

Reply:
`.map(callback)` will pass `element, index` to `callback`, and if the second parameter is `number`, `t.assertIdentifier(labelId)` should throw an error.
I investigated and it seems that this is dead code.